src/clj/fluree/db/api/query.cljc

```diff
@@ -152,11 +166,16 @@
   (go-try
     (try*
       (let [[alias explicit-t] (extract-query-string-t alias)
-            address (<? (nameservice/primary-address conn alias nil))
-            ledger  (<? (jld-ledger/load conn address))
-            db      (ledger/-db ledger)
-            t*      (or explicit-t t)]
-        (<? (restrict-db db t* context opts)))
+            address  (<? (nameservice/primary-address conn alias nil))
+            ledger   (<? (jld-ledger/load conn address))
+            db       (ledger/-db ledger)
+            t*       (or explicit-t t)
+            rules-db (let [dbs-or-aliases (:reasoner-rules-db opts)]
+                       (if (string? (first dbs-or-aliases))
+                         [(ledger/-db (<? (jld-ledger/load conn (first dbs-or-aliases))))]
+                         dbs-or-aliases))
```

Comment:
Here it is assumed there is a valid db object/var. I think this is the right point to ensure that is the case or throw an exception